test/heat_eqn.jl

```diff
@@ -33,7 +33,7 @@ end
     dx = 2π/(N-1)
     x = collect(-pi : dx : pi)
     u0 = @. -(x - 0.5)^2 + 1/12
-    B = CenteredDifference(1,2,dx,N)
+    B = CenteredDifference(1,2,dx,N-2)
```

Comment:
why the size change?

Reply:
> why the size change?

Its operating on a Vector with 512 grid points(including boundaries).
That was working somehow on Centered difference but Upwind was returning dimension mismatch error.